src/main.cc

```diff
@@ -896,18 +905,13 @@ mainReconfigureFinish(void *)
     if (Config2.onoff.enable_purge)
         Config2.onoff.enable_purge = 2;
 
-    // parse the config returns a count of errors encountered.
     const int oldWorkers = Config.workers;
+
     try {
-        if (parseConfigFile(ConfigFile) != 0) {
-            // for now any errors are a fatal condition...
-            self_destruct();
-        }
+        Configuration::Parse();
     } catch (...) {
         // for now any errors are a fatal condition...
-        debugs(1, DBG_CRITICAL, "FATAL: Unhandled exception parsing config file. " <<
-               " Run squid -k parse and check for errors.");
-        self_destruct();
+        fatal(ConfigurationFailureMessage().c_str());
```

Comment:
This PR replaces self_destruct() with fatal() in this reconfiguration error processing context because self_destruct() implementation does not work well outside of parseOneConfigFile(), where the parsing state it expects either does not exist or is invalid/stale. There are other places in official code where self_destruct() is called too early or too late, so this change does not address all such problems, but it does improve things.

self_destruct() calls fatal() after trying to report the current parsing state, so this PR change just removes that reporting from this context (because that reporting does not work in this context).

fatal() itself has serious problems, but they are not specific to parsing, and fixing them is outside this PR scope.

Finally, this change does not affect the vast majority of configuration parsing errors because they are handled inside parseOneConfigFile() (and functions it calls), where self_destruct() does work OK. This PR does not change that code.

Reply:
> Yay! you finally agree with me on this.

I suspect I have never disagreed with you on any relevant "this", whatever this "this" is, but I am glad we agree that this change moves code in the right direction.